apps/job-service/src/jobs/delayed-jobs.controller.ts

```diff
@@ -97,19 +97,23 @@ export class DelayedJobsController {
       },
     },
   })
+  @JsonApiResponse({ data: { type: DelayedJobDto } })
   @ApiException(() => UnauthorizedException, { description: 'Authentication failed.' })
   @ApiException(() => BadRequestException, { description: 'Invalid payload or IDs provided.' })
   @ApiException(() => NotFoundException, { description: 'One or more jobs specified in the payload could not be found.' })
   async bulkClearJobs(
-    @Body() bulkClearJobsDto: JobBulkUpdateBodyDto,
+    @Body() bulkClearJobsDto: DelayedJobBulkUpdateBodyDto,
     @Request() { authenticatedUserId }
-  ): Promise<{ data: JobData[] }> {
+  ): Promise<JsonApiDocument> {
     const jobUpdates = bulkClearJobsDto.data;
-
+  
     if (!jobUpdates || jobUpdates.length === 0) {
       throw new BadRequestException('No jobs provided in the payload.');
     }
-
+  
+    if (!authenticatedUserId) {
+      throw new UnauthorizedException('Authentication failed.');
+    }
```

Comment:
This is done automatically by middleware by default. FYI for the future, you have to use @NoBearerAuth to disable it.

Reply:
removed